src/box/lua/feedback_daemon.lua

```diff
@@ -348,6 +363,73 @@ local function feedback_loop(self)
     self.shutdown:put("stopped")
 end
 
+-- Returns nil if there is no appropriate metrics module.
+-- Otherwise, returns metrics.collect result.
+local function collect_default_metrics()
+    -- Do all the job in pcall for better reliability.
+    local has_metrics, metrics = pcall(function()
+        local metrics = require("metrics")
+        -- Required version of metrics module is 0.16.0 or newer.
+        -- A little cheat here - _VERSION field was introduced in 0.16.0,
+        -- let's just check if it is not nil.
+        if metrics._VERSION == nil then
+            return nil
+        end
+        return metrics.collect({invoke_callbacks = true, default_only = true})
+    end)
+    if has_metrics then
+        return metrics
+    else
+        return nil
+    end
+end
+
+-- Trivial objects are stored right in table slot or stack with gc64
+local trivial_obj_size = ffi.abi('gc64') and 0 or 8
+
+-- Calculate approximate amount of occupied memory
+local function obj_approx_size(obj)
+    if type(obj) == 'table' then
+        local size = 40
+        for k, v in pairs(obj) do
+            size = size + 16
+            size = size + obj_approx_size(k)
+            size = size + obj_approx_size(v)
+        end
+        return size
+    elseif type(obj) == 'string' then
+        return 17 + #obj
+    else -- Number, boolean and nil
+        return trivial_obj_size
+    end
+end
```

Comment:
Nit, feel free to resolve and ignore: I would open a ticket for refactoring this part into `src/lua/utils.h`, as this may be useful in other cases.